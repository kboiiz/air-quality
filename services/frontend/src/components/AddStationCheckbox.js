const AddStationCheckbox = (props) => {
  return (
    <div className="dropdown-input-wrap">
      <label>
        <input
          type="checkbox"
          name={props.name}
          checked={props.checked}
          onChange={props.onChange}
        />
        <span>{props.value}</span>
      </label>
      <ul>
        {props.pollutants ? (
          <ul>
            {props.pollutants.map((pollutant) => (
              <li key={pollutant}>{pollutant}</li>
            ))}
          </ul>
        ) : null}
      </ul>
    </div>
  );
};

export default AddStationCheckbox;
