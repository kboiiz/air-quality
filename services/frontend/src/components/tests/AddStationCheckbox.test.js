import React from "react";
import { BrowserRouter } from 'react-router-dom'
import { getByLabelText, render, screen } from '@testing-library/react'
import "@testing-library/jest-dom"
import userEvent from "@testing-library/user-event"
import AddStationCheckbox from "../AddStationCheckbox";

const onChangeMock = jest.fn()

describe("AddStationCheckbox", () => {
  it("renders a checkbox input with the given name", () => {
    const { container } = render(
      <AddStationCheckbox
        key={"01234"}
        name={"01234"}
        pollutants={["pm25_aqi", "pm10_aqi"]}
        value={"Santa's Workshop"}
        checked={true}
        onChange={onChangeMock}
      />
    )
    expect(container).toMatchSnapshot()
    expect(screen.getByRole("checkbox")).toBeChecked()
  })
})
