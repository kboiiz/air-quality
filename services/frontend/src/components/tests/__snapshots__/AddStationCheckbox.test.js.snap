// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddStationCheckbox renders a checkbox input with the given name 1`] = `
<div>
  <div
    class="dropdown-input-wrap"
  >
    <label>
      <input
        checked=""
        name="01234"
        type="checkbox"
      />
      Santa's Workshop
    </label>
    <ul
      class="add-station-pollutant"
    >
      <li
        class="add-station-pollutant"
      >
        pm25
      </li>
      <li
        class="add-station-pollutant"
      >
        pm10
      </li>
    </ul>
  </div>
</div>
`;
